### PR TITLE
Add taurus recipe

### PR DESCRIPTION
**Pull request by beenje**
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [X] Source is from official source
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [X] If static libraries are linked in, the license of the static library is packaged.
- [X] Build number is 0
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by beenje** [comment]
@cpascual here is a draft PR for taurus.

Could you please confirm you want to be a maintainer.

I split the package as taurus-core and taurus-qt.
I based the requirements on the `setup.py` file. I see there were more dependencies in your [meta.yaml](https://github.com/taurus-org/taurus/blob/develop/conda/meta.yaml). Should some be added?

**Comment by cpascual** [comment]
I confirm that I agree to be a maintainer of this package

**Comment by cpascual** [comment]
> I split the package as taurus-core and taurus-qt.

Nice! 
Is it possible to add a third one called `taurus` that is a metapackage just depending on the other two?

> I based the requirements on the setup.py file. I see there were more dependencies in your meta.yaml. Should some be added?

I checked the differences and I think the ones you used are ok:
- `guidata` and `pythonqwt` should be already dependencies of `guiqwt `. 
- `pymca` is used but not essential, so I guess it can be left out in order to not pull all its (many) dependencies, and let users install it if they want
- I thought `pillow` was needed, but I cannot find any import of `PIL` in the code, so I guess we don't need it
- same for `scipy`

**Comment by beenje** [comment]
> > I split the package as taurus-core and taurus-qt.
> 
> Nice!
> Is it possible to add a third one called `taurus` that is a metapackage just depending on the other two?

Yes, that's already the case. I should have mentioned it. This is done by (taurus is the name of the main package):

```
requirements:
   run:
     - {{ pin_subpackage('taurus-core', max_pin="x.x.x.x") }}
     - {{ pin_subpackage('taurus-qt', max_pin="x.x.x.x") }}
```

If the requirements are OK, I guess this is ready for review then?

**Comment by cpascual** [comment]
> If the requirements are OK, I guess this is ready for review then?

Yes. Thanks a lot for taking care of this!

**Comment by cpascual** [comment]
Regarding the failure in the windows build, is that something to be addressed at the conda recipe level or at taurus level?

**Comment by beenje** [comment]
> Regarding the failure in the windows build, is that something to be addressed at the conda recipe level or at taurus level?

I think it's a recipe or CI issue.
I thought `noarch: python` recipes were only built on Linux and then converted for other platforms.

@conda-forge/help-python, could you please confirm if this is a CI issue or something to change in the recipe?

**Comment by beenje** [comment]
@cpascual, the current recipe can't be built on Windows as I use a bash script `build_core.sh`. I had to use an external script with `outputs` because I didn't manage to use `script: "{{ PYTHON }} -m pip install . -vv"`.
But `python: noarch` recipes should only be built on Linux. The failure on Windows shouldn't matter for this PR.

The conda-forge team will confirm or not :-)

File: recipes/taurus/build_core.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
```
Comment (dopplershift) [inline recipes/taurus/build_core.sh:1]: You can remove this

File: recipes/taurus/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 5329036608b94444f3512c89d493092fb658d069bcd4d1f82f393e70fdefc3d9
+
+build:
+  number: 0
+  noarch: python
```
Comment (dopplershift) [inline recipes/taurus/meta.yaml:14]: Add this here instead:
```
  script: {{ PYTHON }} -m pip install . --no-deps -vv
```

**Review by dopplershift — CHANGES_REQUESTED** [review]
In general, `noarch` packages only need to build on Linux. For staged-recipes, all recipes all built on all 3 platforms--but we can ignore failing builds when it makes sense.

File: recipes/taurus/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+requirements:
+  run:
+    - {{ pin_subpackage('taurus-core', max_pin="x.x.x.x") }}
+    - {{ pin_subpackage('taurus-qt', max_pin="x.x.x.x") }}
+
+outputs:
+  - name: taurus-core
+    script: build_core.sh
```
Comment (dopplershift) [inline recipes/taurus/meta.yaml:23]: script: {{ PYTHON }} -m pip install . --no-deps -vv

Comment (beenje) [reply recipes/taurus/meta.yaml:23]: I tried that before but got:
```
Attempting to finalize metadata for taurus-core
INFO conda_build.metadata:finalize_outputs_pass(748): Attempting to finalize metadata for taurus-core
Error: Failed to render jinja template in /home/conda/staged-recipes-copy/recipes/taurus/meta.yaml:
'PYTHON' is undefined
```

This is why I used an external script.

Comment (beenje) [reply recipes/taurus/meta.yaml]: I just pushed that change anyway to check this wasn't an issue on my machine (even if I used `./.scripts/run_docker_build.sh`).

Comment (beenje) [reply recipes/taurus/meta.yaml]: Same error. Am I doing something wrong?
Or should that be reported as a conda-build bug?

Comment (beenje) [reply recipes/taurus/meta.yaml:23]: I reverted my change to go back to my initial commit. Seems like a file has to be used for `script` under `outputs`.

**Comment by beenje** [comment]
Thanks for the review @dopplershift! Do you agree with my comments or do you expect other changes?

**Review by dopplershift — APPROVED** [review]
Odd that it didn't work as I expected it, but I guess the current setup is fine.

**Comment by beenje** [comment]
Thanks!

**Pull request merged**